src/rulesets/Base/array.jl

```diff
@@ -43,32 +51,81 @@ function rrule(::typeof(Base.vect), X::Vararg{Any,N}) where {N}
     return Base.vect(X...), vect_pullback
 end
 
+"""
+    _make_real_zeros(xdots, xs)
```

Comment:
perhaps `instantiate_zeros`
`make_real_zeros` could be interpretted as a Real vs Complex.

Reply:
Just saw this. For now it's `_instantiate_zeros`, clearly internal, and unlike materialise I can reliably spell it...